spec/app_support/auto_expire_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe AutoExpire, :timecop_freeze do
+  let(:now) { DateTime.now.change(hour: 9, min: 31)  }
+
+  let(:action) { described_class.new }
+  let(:order_detail) { reservation.order_detail }
+
+  describe '#perform' do
+    context 'a new reservation' do
+      let!(:reservation) { create(:purchased_reservation, :yesterday, actual_start_at: 1.hour.ago) }
+
+      it 'completes the reservation' do
+        expect { action.perform }.to change { order_detail.reload.state }.from('new').to('complete')
```

Comment:
We should add some additional checks here like setting the fulfilled_at, that it's a problem reservation, it doesn't get actual pricing assigned,